vendor/github.com/OpenBazaar/wallet-interface/wallet.go

```diff
@@ -51,6 +51,11 @@ type Wallet interface {
 	walletCanBumpFee
 }
 
+var (
+	// ErrInsufficientFunds is returned when the wallet is unable to send the amount specified due to the balance being too low
+	ErrInsufficientFunds = errors.New("ERROR_INSUFFICIENT_FUNDS")
```

Comment:
We should be very thoughtful before prescribing an error sentinel as part of the interface. This prescription requires all wallets which implement the interface to also be aware of and properly return this specific error. I don't see what's producing this error which makes it hard to justify where it should go, but seeing that it's consumed within `core`, that seems like the best place for it's declaration.

Reply:
The rationale here is that during multisig, the wallet would try to submit a txn and in case of eth, there is a distinct possibility that the existing wallet balance may be insufficient to fund the gas required for the txn. In such a case, `ErrInsufficientFunds` will be returned by Multisign. If there is a cleaner solution for wallet implementation and core to share, please let me know.

Reply:
I don't see where `Multisign` is actually returning `ErrInsufficientFunds`. I also don't know why a signing function would be concerned with wallet balance. In my mind, a transaction should be valid going into `Multisign` and should be considering the `ErrInsufficientFunds` scenario well before that point. This seems like a leaky abstraction that should be left to the wallet consumer to handle. Maybe we should discuss this particulars of this before I reject it completely, but I'm not seeing any benefit from this approach.

Reply:
> I also don't know why a signing function would be concerned with wallet balance. 

This applies to ETH as the Multisign function will call the `execute` function in the smart contract, and that function call requires gas to call, and if you wallet doesn't have enough gas it will fail.

But is your point is that we're checking the balance too late and we should fail early for this particular scenario?

Reply:
I'm not sure if there's an earlier opportunity to check/fail for this case. My argument is whether it belongs as part of a signing function which feels like it should have separate responsibilities from balance checking. 

I don't suppose I have enough information to make a good design decision here so I'll push this off for the time being.